test/integration/controlplane/api_all_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlplane
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	apitesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+// Test verifies that the apiserver bootstrap does not regress when all feature
+// gates are enabled.
+// It acts as a sentinel using the integration framework timeouts, if this test fails
+// is an indication that any test on the integration is susceptible to fail because of
+// the apiserver not be able to bootstrap in time.
+// The existing timeouts are:
+// * 1 minute for apiserver to be ready
+// * 30 seconds for default namespace to be created
+// * 30 seconds for kubernetes.default service to be created
+func TestAllAPIsByDefault(t *testing.T) {
```

Comment:
was this test effective at flushing out the issue?

Reply:
yes, is how I reproduced the problem locally, otherwise it was impossiblel to get those stack traces